bundles/http/Citrix/7150aa4a-3bcd-4a73-a15c-893249515184/101b58cb-1023-48a1-9aca-4b64c2dbfd68.js

```diff
@@ -0,0 +1,412 @@
+      } else {
+        console.log('WARNING: Endpoint views - Bad or empty response')
+      }
+
+    } catch (error) {
+      console.error(error)
+    }
+
+  } while (pageSize * pageNumber < totalAvailable)
+}
+
+async function syncWorkbooks ({ client, dataStore, credentials }) {
+  let pageNumber = 0
+  let totalAvailable
+
+  do {
+    pageNumber++
+
+    const body = await request(client, 'workbooks', credentials, pageNumber)
+    
+    try {
+      const workbooks = body.workbooks.workbook
+      totalAvailable = validateTotalAvailable(body, 'workbooks')
+
+      if (Array.isArray(workbooks) && workbooks.length > 0) {
+        workbooks.forEach(workbook => { // flaten nested JSONs
+          flatJson(workbook, 'owner')
+          flatJson(workbook, 'project')
+
+          // convert showTabs to boolean
+          workbook.showTabs = workbook.showTabs === 'true'
+
+          // convert size to integer
+          const workbookSize = parseInt(workbook.size)
+          if (!isNaN(workbookSize)) {
+            workbook.size = workbookSize
+          } else {
+            delete workbook.size
+          }
+        })
+        dataStore.save('workbooks', workbooks)
+      } else {
+        console.log('WARNING: Endpoint workbooks - Bad or empty response')
+      }
+
+    } catch (error) {
+      console.error(error)
+    }
+
+  } while (pageSize * pageNumber < totalAvailable)
+}
+
+async function getCredentials (client, params) {
+  const request = {
+    credentials: {
+      name: `${params.username}`,
+      password: `${params.password}`,
+      site: {
+        contentUrl: `${params.site}`
+      }
+    }
+  }
+
+  console.log("authorization request sent")
+
+  let response = await client.fetch('/api/3.7/auth/signin', {
+    method: 'POST',
+    body: JSON.stringify(request),
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  })
+
+  if (!response.ok) {
+    throw new Error(`Auth Request failed(${response.status}: ${response.statusText})`)
+  }
+
+  response = await response.json()
+
+  console.log("authorization finished")
+
+  return response.credentials
+}
+
+async function request(client, entity, credentials, pageNumber) {
+  console.log('sync of ' + entity + ' page ' + pageNumber + ' started')
+
+  let path =`api/3.7/sites/${credentials.site.id}/${entity}?pageSize=${pageSize}&pageNumber=&${pageNumber}` 
+  
+  if (entity === "users") { 
+    path = `${path}&fields=_all_`; 
+  }
+
+  const response = await client.fetch(path,
+    {
+      headers: {
+        'X-Tableau-Auth': credentials.token
+      }
+    })
+
+  if (!response.ok) {
+    console.log(`WARNING: Endpoint ${entity} - Request failed(${response.status}: ${response.statusText})`)
+    //throw new Error(`Endpoint ${entity} - Request failed(${response.status}: ${response.statusText})`)
+  }
+  console.log(entity + ' page ' + pageNumber + ' - response received, status: ' + response.status)
+
+  return await response.json()
```

Comment:
If JSON is always present, event in the error response (so you don't need a better stacktrace) then you can remove `await`
```suggestion
  return response.json()
```
More info [here](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md)